core/trino-spi/src/main/java/io/trino/spi/function/LanguageFunction.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.function;
+
+import io.trino.spi.connector.CatalogSchemaName;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public record LanguageFunction(
+        String signatureToken,
```

Comment:
It's an opaque token that represents the signature for the purposes of uniquely identifying the overload variants of functions with the same name. We don't call it "signature" because plugins should not be making any assumptions about the format or otherwise using it.